utils/utils.jsx

```diff
@@ -1187,18 +1189,18 @@ export function getDisplayName(user) {
 /**
  * Gets the display name of the user with the specified id, respecting the TeammateNameDisplay configuration setting
  */
-export function getDisplayNameByUserId(userId, usernameWithPrefix = false) {
-    return getDisplayNameByUser(UserStore.getProfile(userId), usernameWithPrefix);
+export function getDisplayNameByUserId(userId) {
+    return getDisplayNameByUser(UserStore.getProfile(userId));
 }
 
 /**
  * Gets the display name of the specified user, respecting the TeammateNameDisplay configuration setting
  */
-export function getDisplayNameByUser(user, usernameWithPrefix = false) {
+export function getDisplayNameByUser(user) {
     const state = store.getState();
     const teammateNameDisplay = getTeammateNameDisplaySetting(state);
     if (user) {
-        return displayUsername(user, teammateNameDisplay, usernameWithPrefix);
```

Comment:
Should we continue to explicitly pass `false` here?

Reply:
`usernameWithPrefix` is added for an entirely different reason which is not used anymore. So I removed it. 

By default, we do get someone as username and used in most places such as typing or even DM header. Which is  expected and for now we are forcing the loader only in LHS

Reply:
No objection to not passing `usernameWithPrefix`, but in https://github.com/mattermost/mattermost-webapp/pull/1502/files#diff-aae7e23473b48cdca24868938c351b6eR84 you changed it to explicitly pass `false` as the third parameter:
```js
            channelDisplayName = displayUsername(teammate, teammateNameDisplay, false);
```

I'm wondering if we should do the same here for consistency?

Reply:
Oh, I also see your new PR to change the semantics of this third parameter now. That makes more sense, though I've never been a fan of opaque `true` and `false` parameters. I wonder if a new function like `displayUsernameWithDefaultFallback` would be clearer?

Reply:
I am not a fan of true, false either if they are beign sent as arguments. 
I would like an object `{user, useDefaultUserName:, displayPreference} ` like this so it is more obvious when we call it

> displayUsernameWithDefaultFallback

It does makes sense but i still prefer object version rather than two functions. `UsernameWithDefaultFallback`, `displayUsername`

I did think about changing it to object but that was too big a PR as it will effect lot of places including mobile repo

Reply:
are we sure the prefix is not used anymore, not even in mobile? what about people already using this function with `mattermost-redux` this seems to be a breaking change to me

Reply:
This is not used anymore in our repos. It was added for grouping system messages Later it is removed for profile popovers. 

Regarding someone using it from matermos-redux I can add new function if you are strong on the opinion.

i am 1/5 that devs depending on this as this was introduced fairly recently and not passing the third param still resorts to the older behaviour